userapi/src/main/java/fk/prof/userapi/api/AggregatedProfileLoader.java

```diff
@@ -42,14 +52,19 @@ public void load(Future<AggregatedProfileInfo> future, AggregatedProfileNamingSt
 
         InputStream in = new StorageBackedInputStream(asyncStorage, filename);
 
+        ProcessGroupTag pgTag = new ProcessGroupTag(filename.appId, filename.clusterId, filename.procId);
```

Comment:
load(), loadSummary(), loadFromInputStream(), loadSummaryFromInputStream() are all executing synchronously. Then why are they taking future as input and populating the same. This is unnecessarily confusing and gives a wrong idea. 
Can they not be just straightforward methods which return a value, instead of populating the future method param with return value.

If where they are called from happens in an async fashion, then future can be wrapped in the caller of these methods

Reply:
Refer to ProfileStoreApiImpl.java:171. This is how these methods are invoked.
But I agree, now with fresh set of eyes :) it does look wierd. 
Delegating the work to worker pool can be done inside. And these methods can directly return a Future.